dashboard-ui/src/pages/home.tsx

```diff
@@ -274,6 +328,33 @@ const KubeContextPicker = ({
   );
 };
 
+/**
+ * SearchBox component
+ */
+
+const SearchBox = () => {
+  const { setSearch } = useContext(Context);
+  const deboucedSearch = useDebounceCallback((value: string) => setSearch(value), 300);
+
+  return (
+    <div
+      className={cn(
+        'search-input relative',
+      )}
+    >
+      <FormControl
+        id="search-box"
+        onChange={(e) => deboucedSearch(e.target.value)}
+        type="search"
+        onKeyDown={(e) => { if (e.key === 'Enter') { e.preventDefault(); } }} // prevent form submission on enter
```

Comment:
If you wrap `<FormControl>` in a `<Form onSubmit={(ev) => ev.preventDefault}>` you can avoid using `onKeyDown`.